apps/studio/components/interfaces/RoleImpersonationSelector/RoleImpersonationSelector.tsx

```diff
@@ -0,0 +1,133 @@
+  )
+
+  function onSelectedChange(value: PostgrestRole) {
+    if (value === 'service_role') {
+      // do not set a role for service role
+      // as the default role is the "service role"
+      state.setRole(undefined)
+    }
+
+    if (value === 'anon') {
+      state.setRole({
+        type: 'postgrest',
+        role: value,
+      })
+    }
+
+    setSelectedOption(value)
+  }
+
+  return (
+    <>
+      <div className={cn('flex flex-col gap-3', padded ? 'p-5' : 'pb-5')}>
+        <h2 className="text-foreground text-base">Database connection settings</h2>
+
+        <form
+          onSubmit={(e) => {
+            // don't allow form submission
+            e.preventDefault()
+          }}
+        >
+          <fieldset className="flex gap-3">
+            <RoleImpersonationRadio
+              value="service_role"
+              isSelected={selectedOption === 'service_role'}
+              onSelectedChange={onSelectedChange}
+              label={serviceRoleLabel}
+              icon={<ServiceRoleIcon isSelected={selectedOption === 'service_role'} />}
+            />
+
+            <RoleImpersonationRadio
+              value="anon"
+              isSelected={selectedOption === 'anon'}
+              onSelectedChange={onSelectedChange}
+              icon={<AnonIcon isSelected={selectedOption === 'anon'} />}
+            />
+
+            <RoleImpersonationRadio
+              value="authenticated"
+              isSelected={
+                selectedOption === 'authenticated' &&
+                (isAuthenticatedOptionFullySelected || 'partially')
+              }
+              onSelectedChange={onSelectedChange}
+              icon={<AuthenticatedIcon isSelected={selectedOption === 'authenticated'} />}
+            />
+          </fieldset>
+        </form>
+
+        {selectedOption === 'service_role' && (
+          <p className="text-foreground-light text-sm">
+            The default Postgres/superuser role. This has admin privileges.
+            <br />
+            It will bypass Row Level Security (RLS) policies.
+          </p>
+        )}
+
+        {selectedOption === 'anon' && (
+          <p className="text-foreground-light text-sm">
+            For "anonymous access". This is the role which the API (PostgREST) will use when a user
+            <br />
+            is not logged in. It will respect Row Level Security (RLS) policies.
+          </p>
+        )}
+
+        {selectedOption === 'authenticated' && (
+          <p className="text-foreground-light text-sm">
+            For "authenticated access". This is the role which the API (PostgREST) will use when
+            <br /> a user is logged in. It will respect Row Level Security (RLS) policies.
+          </p>
+        )}
+      </div>
+
+      {selectedOption === 'authenticated' && (
+        <>
+          <DropdownMenuSeparator />
+          <div
+            className={cn('pt-5', isAuthenticatedOptionFullySelected && 'pb-5', padded && 'px-5')}
```

Comment:
needs some padding at the bottom?
![image](https://github.com/supabase/supabase/assets/19742402/12d0d5d4-ba21-4eef-aa18-4e82b06e68cb)

Reply:
I did it that way because with more than one user I think it looks better overflowing directly on the bottom, but it is bad for < 4 users, so added padding at all times!

<img width="664" alt="Screenshot 2023-11-30 at 17 50 10" src="https://github.com/supabase/supabase/assets/10985857/b65e7ab7-4eb7-48e1-beb7-f5950240ca57">